source/agora/network/Manager.d

```diff
@@ -994,25 +995,42 @@ public class NetworkManager
     public void getUnknownTXs (Ledger ledger) @safe nothrow
     {
         auto unknown_txs = ledger.getUnknownTXHashes();
+        log.trace("getUnknownTXs: detected {} unknown txs", unknown_txs.length);
 
         foreach (peer; this.peers[])
         {
             if (unknown_txs.length == 0)
                 break;
 
-            foreach (tx; peer.client.getTransactions(unknown_txs))
+            // Fetch transactions in chunks
+            auto hashes = unknown_txs[].map!((Hash h) => h).array; // can not be lazy so use array
+            auto added = hashes.chunks(8)
```

Comment:
Why did we have to limit the TXs per request, were we hitting a timeout or something?

Reply:
When I reproduced it locally I found that limiting the number in the Set to 8 enabled the call via vibe.d to actually happen.

Reply:
Sounds like a workaround to me but it might not be worth to dive deep into vibe.d for this.

Reply:
The long term goal is to switch to RPC and use ranges so not worth the time at the moment.